### PR TITLE
Update and improve README plus fix version references

### PR DESCRIPTION
**Pull request by miguno**
Various fixes, improvements, and clarifications for the top-level README.

### CODE REVIEW

**Comment by cadonna** [comment]
Hi @miguno,

Today I run the `WordCount` example and when I followed the instructions in the README I found the following errors:

- `Packaging and Running the examples`:
    - `$ ./bin/zookeeper-server-start ./etc/kafka/zookeeper.properties` 
         should be 
       `$ ./bin/zookeeper-server-start.sh ./config/zookeeper.properties`
   -  `$ ./bin/kafka-server-start ./etc/kafka/server.properties`
         should be
       `$ ./bin/kafka-server-start.sh ./config/server.properties`

Furthermore, I found a couple of missing `.sh` in the instructions in the header in `src/main/java/io/confluent/examples/streams/WordCountLambdaExample.java`.

Do you want to fix this errors in this PR or do you want me to create a new PR. With a new PR, we might risk to run into merge conflict.

**Comment by miguno** [comment]
@cadonna : This would be a separate PR. However, I think what you are seeing is the difference between the names of these command line tools in Confluent Platform (without `.sh`) and Apache Kafka (with `.sh` suffix). The instructions in the examples of this repo assume the use of Confluent Platform.

**Comment by miguno** [comment]
See https://github.com/confluentinc/kafka-streams-examples/blob/5.2.1-post/src/main/java/io/confluent/examples/streams/WordCountLambdaExample.java#L57:

```
* Note: The above commands are for the Confluent Platform. For Apache Kafka it should be {@code bin/kafka-topics.sh ...}.
```

File: docker-compose.yml [hunk 1]
```diff
@@ -88,7 +88,7 @@ services:
                        cub kafka-ready -b kafka:29092 1 20 && \
                        echo Waiting for Confluent Schema Registry to be ready... && \
                        cub sr-ready schema-registry 8081 20 && \
-                       java -cp /usr/share/java/kafka-streams-examples/kafka-streams-examples-5.0.0-standalone.jar \
+                       java -cp /usr/share/java/kafka-streams-examples/kafka-streams-examples-5.2.1-standalone.jar \
```
Comment (miguno) [inline docker-compose.yml]: @JimGalasyn : I think this is the right path and filename (`/usr/share/java/kafka-streams-examples/kafka-streams-examples-5.2.1-standalone.jar`) but I haven't had the time to actually verify the Docker packaging/setup -- I just searched and replaced for outdated versions/filenames for the packaged jar.

File: docs/index.rst [hunk 2]
```diff
@@ -145,15 +145,15 @@ Running further Confluent demo applications for the Kafka Streams API
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The container named ``kafka-music-application``, which runs the Kafka Music demo application, actually contains all of
-Confluent's `Kafka Streams demo applications <https://github.com/confluentinc/examples>`__.  The demo applications are
-packaged in the fat jar at ``/usr/share/java/kafka-streams-examples/kafka-streams-examples-4.0.0-standalone.jar`` inside this container.
+Confluent's `Kafka Streams demo applications <https://github.com/confluentinc/kafka-streams-examples>`__.  The demo applications are
+packaged in the fat jar at ``/usr/share/java/kafka-streams-examples/kafka-streams-examples-5.2.1-standalone.jar`` inside this container.
```
Comment (JimGalasyn) [inline docs/index.rst]:
```suggestion
packaged in the fat jar at ``/usr/share/java/kafka-streams-examples/kafka-streams-examples-|release|-standalone.jar`` inside this container.
```

File: docs/index.rst [hunk 3]
```diff
@@ -145,15 +145,15 @@ Running further Confluent demo applications for the Kafka Streams API
...
 This means you can easily run any of these applications from inside the container via a command similar to:
 
 .. sourcecode:: bash
```
Comment (JimGalasyn) [inline docs/index.rst]:
```suggestion
.. codewithvars:: bash
```

File: docs/index.rst [hunk 4]
```diff
@@ -145,15 +145,15 @@ Running further Confluent demo applications for the Kafka Streams API
...
 
     # Example: Launch the WordCount demo application (inside the `kafka-music-application` container)
     $ docker-compose exec kafka-music-application \
-            java -cp /usr/share/java/kafka-streams-examples/kafka-streams-examples-4.0.0-standalone.jar \
+            java -cp /usr/share/java/kafka-streams-examples/kafka-streams-examples-5.2.1-standalone.jar \
```
Comment (JimGalasyn) [inline docs/index.rst]:
```suggestion
            java -cp /usr/share/java/kafka-streams-examples/kafka-streams-examples-|release|-standalone.jar \
```

File: docs/index.rst (hunk 2 above)
Comment (JimGalasyn) [inline docs/index.rst]:
```suggestion
packaged in the fat jar at ``/usr/share/java/kafka-streams-examples/kafka-streams-examples-|release|-standalone.jar`` inside this container.
```

**Comment by miguno** [comment]
pint-merged as well

**Pull request merged**